### ISSUE TITLE
using `gr.LogoutButton` breaks app

### ISSUE THREAD

**Issue by xianbaoqian** [gradio-app/gradio#7695]
### Describe the bug

https://www.gradio.app/guides/sharing-your-app#o-auth-login-via-hugging-face needs to be updated

Copy & pasting the code causes an error after the app is started.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```python
import gradio as gr

```

### Screenshot

<img width="1244" alt="image" src="https://github.com/gradio-app/gradio/assets/38108242/458723da-320f-4309-8fe0-4298d7abfe23">

### System Info

```shell
default gradio version on HF space
```

### Severity

I can work around it

**Comment by pngwn** [comment]
This is a deprecation so shouldn't break anything.

@Wauplin When we unified the login and logout button, did we break the previous behaviour?

**Comment by pngwn** [comment]
Relevant PR: https://github.com/gradio-app/gradio/pull/7063

**Comment by Wauplin** [comment]
Thanks for the ping @pngwn. `gr.LogoutButton` is indeed deprecated but that's not the problem here (it raises a UserWarning but that's all). The snippet in the docs is missing the `demo.launch()` line, that's why no errors is raised in the logs (+ also a `huggingface_hub` import). I opened https://github.com/gradio-app/gradio/pull/7697 to fix this.

I also realized OAuth is currently broken in the latest version of gradio. Fixed that too in the PR (more details there).
